misc/loop/cmd/main.go

```diff
@@ -0,0 +1,235 @@
+func (s *service) startPortalLoop(ctx context.Context) error {
+	logrus.Info("Starting the Portal Loop")
+
+	// 1. Pull latest docker image
+	isNew, err := s.portalLoop.pullLatestImage(ctx)
+	if err != nil {
+		return err
+	}
+
+	// 2. Get existing portal loop
+	containers, err := s.portalLoop.getPortalLoopContainers(ctx)
+	if err != nil {
+		return err
+	} else if len(containers) == 0 {
+		logrus.Info("No portal loop instance found, starting one")
+		// Portal loop isn't running, Starting it
+		container, err := s.portalLoop.startPortalLoopContainer(context.Background())
+		if err != nil {
+			return err
+		}
+		containers = []types.Container{*container}
+
+		for _, p := range container.Ports {
+			if p.Type == "tcp" && p.PrivatePort == uint16(26657) {
+				s.portalLoopURL = fmt.Sprintf("http://localhost:%d", int(p.PublicPort))
+				s.portalLoop.switchTraefikPortalLoop(s.portalLoopURL)
+				break
+			}
+		}
+		return nil
+	} else if s.portalLoopURL == "" {
+		for _, p := range containers[0].Ports {
+			if p.Type == "tcp" && p.PrivatePort == uint16(26657) {
+				s.portalLoopURL = fmt.Sprintf("http://localhost:%d", int(p.PublicPort))
+				s.portalLoop.switchTraefikPortalLoop(s.portalLoopURL)
+				break
+			}
+		}
+	}
+
+	// 3. Check if there is a new image
+	if !isNew {
+		return nil
+	}
+
+	// 4. Set traefik in READ ONLY mode
+	err = s.portalLoop.switchTraefikMode(setReadOnly)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = s.portalLoop.switchTraefikMode(unsetReadOnly)
+		if err != nil {
+			logrus.WithError(err).Error()
+		}
+	}()
+
+	// 5. Backup TXs
+	err = s.portalLoop.backupTXs(ctx)
+	if err != nil {
+		return err
+	}
+
+	// 6. Start a new portal loop
+	container, err := s.portalLoop.startPortalLoopContainer(context.Background())
+	if err != nil {
+		return err
+	}
+	for _, p := range container.Ports {
+		if p.Type == "tcp" && p.PrivatePort == uint16(26657) {
+			s.portalLoopURL = fmt.Sprintf("http://localhost:%d", int(p.PublicPort))
+			break
+		}
+	}
+
+	// 7. Wait 5 blocks new portal loop to be ready
+	now := time.Now()
+	for {
+		if time.Since(now) > time.Second*120 {
+			return fmt.Errorf("timeout getting latest block")
+		}
+		err := func() error {
+			resp, err := http.Get(s.portalLoopURL + "/status")
+			if err != nil {
+				return err
+			}
+			defer resp.Body.Close()
+
+			tmStatus := struct {
+				Result struct {
+					SyncInfo struct {
+						LatestBlockHeight string `json:"latest_block_height"`
+					} `json:"sync_info"`
+				} `json:"result"`
+			}{}
+			if err := json.NewDecoder(resp.Body).Decode(&tmStatus); err != nil {
+				return err
+			}
+
+			currentBlock, err := strconv.Atoi(tmStatus.Result.SyncInfo.LatestBlockHeight)
+			if err != nil {
+				return err
+			} else if currentBlock >= 5 {
+				return nil
+			}
+			return fmt.Errorf("blocks: %d/5", currentBlock)
+		}()
+		if err != nil {
+			if !strings.HasPrefix(err.Error(), "blocks: ") {
+				logrus.WithError(err).Error()
+			}
+		} else {
+			break
+		}
```

Comment:
```suggestion
		if err == nil {
			break
		}
		
		if !strings.HasPrefix(err.Error(), "blocks: ") {
			logrus.WithError(err).Error()
		}
```

btw, you're silently ignoring errors not starting with "blocks".

Reply:
Yes, couple line above, i'm returning this error, which i don't want to be printed to avoid too many logs

```go
    return fmt.Errorf("blocks: %d/5", currentBlock)
```